sensor_msgs/include/sensor_msgs/fill_image.hpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2008, Willow Garage, Inc.
+// All rights reserved.
+//
+// Software License Agreement (BSD License 2.0)
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above
+//    copyright notice, this list of conditions and the following
+//    disclaimer in the documentation and/or other materials provided
+//    with the distribution.
+//  * Neither the name of the Willow Garage nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+// FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+// COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+// INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+// BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+// LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+// ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+// This file is originally ported from ROS1:
+// https://github.com/ros/common_msgs/blob/jade-devel/sensor_msgs/include/sensor_msgs/fill_image.h
+
+#ifndef SENSOR_MSGS__FILL_IMAGE_HPP_
+#define SENSOR_MSGS__FILL_IMAGE_HPP_
+
+#include <sensor_msgs/msg/image.hpp>
+#include <sensor_msgs/image_encodings.hpp>
+
+#include <string>
```

Comment:
I believe system includes should be listed before external third-party headers and project headers. The only exception would be if this were an implementation of a header, you could include that header first. See:

https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

Reply:
We have a test for this in cpplint (`ament_cpplint`), but sadly it doesn't work very well.

Reply:
Yeah cpplint assume anything within `<` to be a system header. The "correct" thing to do here would be:
```c++
#include <string>

#include "sensor_msgs/msg/image.hpp"
#include "sensor_msgs/image_encodings.hpp"
```

I assumed it was using `<` to stay close to original file but apparently the ROS 1 version uses double quotes as well so I think it makes sense to change it here and put the system header first